ee/zentral/contrib/intune/models.py

```diff
@@ -0,0 +1,53 @@
+import logging
+from django.db import models
+from django.db.models import F
+from django.urls import reverse
+
+from zentral.core.secret_engines import decrypt_str, encrypt_str, rewrap
+
+
+logger = logging.getLogger("zentral.contrib.intune.models")
+
+
+class Tenant(models.Model):
+    business_unit = models.ForeignKey("inventory.BusinessUnit", on_delete=models.PROTECT)
+    name = models.CharField(max_length=256, unique=True)
+    description = models.TextField(blank=True, null=True)
+    # Authentication
+    tenant_id = models.CharField(max_length=256, help_text="The microsoft Azure Tenant ID")
+    client_id = models.UUIDField(help_text="The client ID of your app registration")
```

Comment:
Client ID should also have a unique=True constraint. I do not see a case for the same client being used with 2 different tenants, and this could avoid simple config errors.

Reply:
I'll add. Pretty impossible to have duplicated entries as this is the application ID that the [Azure app registration portal](https://go.microsoft.com/fwlink/?linkid=2083908) assigned to the app.